themes/theme-ios11/components/FilterBar/components/Content/components/FilterChips/index.jsx

```diff
@@ -120,18 +128,35 @@ class FilterChips extends Component {
             const removeLabel = i18n.text('filter.remove', { filter: filterFormatted });
             const editLabel = i18n.text('filter.edit', { filter: filterFormatted });
 
-            chips.push((
-              <Chip
-                id={value.id}
-                key={`filter-${value.id}`}
-                onRemove={() => this.handleRemove(filter.id, value.id)}
-                onClick={openFilters}
-                removeLabel={removeLabel}
-                editLabel={editLabel}
-              >
-                {filterFormatted}
-              </Chip>
-            ));
+            if (pattern === CATEGORY_ALL_PATTERN) {
```

Comment:
i'm not happy with this approach. we should try to find a way where the filter key is removed from the `filters` object before it reaches this component.

Reply:
@fkloes So i created a helper to remove the categories filter from the `filter` object, but when i remove the categories filter from the `filter` object in the `FilterBarContent` component (before the filters reach the `FilterChips`), it seems the products get fetched again without the categories filter...

Reply:
I left the helper function in the ne commit

Reply:
since we now have the central `getShowAllProductsFilters` helper which creates the filters object, i just introduced new `isHidden` prop for the filters and parsed them inside the `FilterChips` and `FilterBarContent`.
So i was able to remove some route pattern related logic from the components.